sandbox/cgroup/monitor.go

```diff
@@ -33,122 +33,190 @@ import (
 // inotifyWatcher manages the inotify watcher, allowing to have a single watch descriptor open
 type inotifyWatcher struct {
 	// The watch object
-	wd *inotify.Watcher
+	wd    *inotify.Watcher
+	wdErr error
 	// This is used to ensure that the watcher goroutine is launched only once
 	doOnce sync.Once
+	// Context
+	ctx context.Context
 	// This channel is used to add a new CGroup that has to be checked
 	addWatchChan chan *groupToWatch
 	// Contains the list of groups to monitor, to detect when they have been deleted
 	groupList []*groupToWatch
 	// Contains the list of paths being monitored by the inotify watcher.
-	// The paths monitored aren't the CGroup paths, but the parent ones, because
-	// in /sys/fs is not possible to detect the InDeleteSelf message, only the
-	// InDelete one, so we must monitor the parent folder and detect when a
-	// specific folder, corresponding to a CGroup, has been deleted.
-	// Each path has associated an integer, which is an usage counter. Every
-	// time a new CGroup paths have to be monitored, the parent paths are
-	// checked in the map, and if they already exist, the usage counter is
-	// incremented by one (of course, if a parent path isn't in the map,
-	// it is initialized to one). Every time the path of a CGroup is deleted,
-	// the corresponding usage counter here corresponding to the parent path
-	// is decremented, and when it reaches zero, it is removed and inotify is
-	// notified to stop monitoring it.
-	pathList map[string]int32
+	// The paths monitored are both the actual leaf path and the parent
+	// directory. See discussion in addWatch for details on why the parent
+	// needs to be tracked as well.
+	pathList map[string]uint
+
+	done     func()
```

Comment:
Since this is always a cancel function, should we say:
```suggestion
	done     context.CancelFunc
```

To make this clear?